### PR TITLE
Disable DSA tests in FIPS mode

### PR DESCRIPTION
**Pull request by tiran**
See: #6880

### CODE REVIEW

**Comment by reaperhulk** [comment]
Could you push an empty commit here? codecov is being dumb.

**Comment by reaperhulk** [comment]
Coverage issue looks legit now (some FIPS checks that now don't take the branches since FIPS entirely skips them)

File: tests/x509/test_x509.py [hunk 1]
```diff
@@ -2646,7 +2646,9 @@ def test_sign_ec_with_md5(self, backend):
     def test_build_cert_with_dsa_private_key(
         self, hashalg, hashalg_oid, backend
     ):
-        if backend._fips_enabled and hashalg is hashes.SHA1:
+        if (
+            backend._fips_enabled and hashalg is hashes.SHA1
+        ):  # pragma: no cover
```
Comment (reaperhulk) [inline tests/x509/test_x509.py:2651]: If we always skip FIPS due to the dsa_supported check returning False when FIPS is enabled, this entire branch can be removed, right?

**Comment by tiran** [comment]
I don't understand why CodeCov is complaining about skip class decorator in two out of four places. https://app.codecov.io/gh/pyca/cryptography/compare/6916/tree/tests/hazmat/primitives/test_dsa.py#D1L605

File: tests/hazmat/primitives/test_dsa.py [hunk 2]
```diff
@@ -511,6 +518,10 @@ def test_prehashed_digest_mismatch(self, backend):
             private_key.sign(digest, prehashed_alg)
 
 
+@pytest.mark.supported(
+    only_if=lambda backend: backend.dsa_supported(),
+    skip_message="Does not support DSA.",
+)
```
Comment (alex) [inline tests/hazmat/primitives/test_dsa.py:524]: This isn't necessary, none of these tests touch the backend.

File: tests/hazmat/primitives/test_dsa.py [hunk 3]
```diff
@@ -591,6 +602,10 @@ def test_repr(self):
         )
 
 
+@pytest.mark.supported(
```
Comment (alex) [inline tests/hazmat/primitives/test_dsa.py:605]: Same

Comment (tiran) [reply tests/hazmat/primitives/test_dsa.py]: Fixed

File: tests/hazmat/primitives/test_dsa.py (hunk 2 above)
Comment (tiran) [reply tests/hazmat/primitives/test_dsa.py]: Fixed

**Review by reaperhulk — APPROVED** [review]

**Pull request merged**